src/screens/ForumScreen/ForumScreen.js

```diff
@@ -1,14 +1,146 @@
-import React, {Component} from "react";
-import {View, SafeAreaView} from "react-native";
-import HeaderBar from "../../components/HeaderBar/HeaderBar";
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-alert */
+/* eslint-disable no-undef */
+/* eslint-disable react-native/no-inline-styles */
+import React from "react";
+import {Component} from "react";
+import {
+  View,
+  Text,
+  TouchableOpacity,
+  Image,
+  SafeAreaView,
+  FlatList,
+  Alert,
+} from "react-native";
+import {firebase} from "../../config/config.js";
 import {LOGO} from "../../config/styles.js";
+import styles from "./styles";
 
 class ForumScreen extends Component {
+  state = {
+    dialogvisible: false,
+    forumtext: "",
+    roomArray: [],
+    isAdmin: null,
+    isSelected: false,
+  };
+  componentDidMount() {
+    const {roomArray} = this.state;
+    const userID = firebase.auth().currentUser.uid;
+    firebase
+      .database()
+      .ref()
+      .on("value", snapshot => {
+        newroomArray = [];
+        snapshot.forEach(element => {
+          newroomArray.push(element.key);
+          this.setState({roomArray: newroomArray});
+        });
+      });
+    firebase
+      .firestore()
+      .collection("users")
+      .doc(userID)
+      .get()
+      .then(doc => {
+        if (doc.exists) {
+          this.setState({isAdmin: doc.data().adminRights});
+        }
+      });
+  }
+
   render() {
+    const {roomArray, isAdmin, isSelected} = this.state;
     return (
       <SafeAreaView>
         <View>
-          <HeaderBar image={LOGO} title={"Forums"} />
+          <View style={{marginBottom: 15}}>
+            <View style={styles.HeaderStyle1}>
+              <Image source={LOGO} style={styles.ImageView} />
+              <Text style={styles.HeaderStyle}>Forums</Text>
+              {isAdmin ? (
+                <TouchableOpacity
+                  style={styles.AskButtonStyle}
+                  onPress={() => {
+                    this.props.navigation.navigate("AddChat");
+                  }}>
+                  <Text
+                    style={{
```

Comment:
Let's move all the styles into the separate styles file of this component

Reply:
done